encord/objects/spaces/base_space.py

```
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+)
+
+from encord.exceptions import LabelRowError
+from encord.objects.spaces.annotation.base_annotation import _ClassificationAnnotation, _ObjectAnnotation
+from encord.objects.spaces.types import SpaceInfo
+from encord.objects.types import (
+    ClassificationAnswer,
+    ObjectAnswer,
+    ObjectAnswerForGeometric,
+    ObjectAnswerForNonGeometric,
+)
+
+if TYPE_CHECKING:
+    from encord.objects import ClassificationInstance, ObjectInstance
+    from encord.objects.ontology_labels_impl import LabelRowV2
+
+SpaceT = TypeVar("SpaceT", bound="Space")
+
+
+class Space(ABC):
+    """
+    Manages the objects on a space within LabelRowV2.
+    Users should not instantiate this class directly, but must obtain these instances via LabelRow.list_spaces().
+    """
+
+    space_id: str
+    _label_row: LabelRowV2
+    _objects_map: dict[str, ObjectInstance]
+    _classifications_map: dict[str, ClassificationInstance]
+
+    def __init__(self, space_id: str, label_row: LabelRowV2):
+        self.space_id = space_id
+        self._label_row = label_row
+        self._objects_map: dict[str, ObjectInstance] = dict()
+        self._classifications_map: dict[str, ClassificationInstance] = dict()
+
+    def get_object_instances(self) -> list[ObjectInstance]:
```

I can now call this method when the row is uninitialized, which I think should throw an error, right? Same for classifications